docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,76 @@
+| Publish project |        n        |           n          |        n       |        n        |           n          |        n       |        n       |          n          |       Y       |          n          |             n            |          n         |        n       |
+| Delete project  |        n        |           n          |        n       |        n        |           n          |        n       |        n       |          n          |       Y       |          n          |             n            |          n         |        Y       |
+| Delete scenario |        n        |           n          |        n       |        n        |           n          |        Y       |        n       |          n          |       n       |          n          |             n            |          n         |        n       |
+| Delete solution |        n        |           n          |        Y       |        n        |           n          |        n       |        n       |          n          |       n       |          n          |             n            |          n         |        n       |
+
+## Association between roles and users for resources
+
+The association between roles and users for resources will be stored in database tables in the `marxan-api` database,
+using one database table for each resource:
+* `users_organizations` will store the roles of users inside organizations.
+* `users_projects` will store the roles of users inside projects.
+* `users_scenarios` will store the roles of users inside scenarios.
+* `users_solutions` will store the roles of users inside solutions.
+
+These tables should contain columns for `user_id`, `role` and the resource id (`organization_id`, `project_id`, `scenario_id` 
+or `solution_id`, respectively).
+
+## Architecture
+
+A main module `access-control` should be created, which will be the main point-of-contact for checking permissions. 
+Additionally, we will have specific modules for checking permissions for each of the resources supported:
+* `organizations-acl`: module that controls access to organizations;
+* `projects-acl`: module that controls access to projects;
+* `scenarios-acl`: module that controls access to scenarios;
+* `solutions-acl`: module that controls access to solutions;
+
+This `access-control` module will expose a public service, and this is the point of entry that should be used by other 
+services to control access to resources. The `access-control` module will also contain the interface that should be 
+implemented the services in the resource-specific modules, which these should implement.
+
+This interface is called `IAccessControlService` and it is located in `api/apps/api/src/modules/access-control/access-control-service.interface.ts`.
```

Comment:
very minor - this looks like the kind of detail that may go out of sync at some point with code (which is hard to avoid in general, and I wouldn't stress about it - we can review things every now and then), and it actually already is out of sync - I'd remove it altogether, and keep the general points about the implementation strategy above.

Reply:
I agree, also the initial access points/interfaces/classes are within the PR there already, so easy to find.

Reply:
Removed this line :+1: